metaflow/plugins/kfp/kfp_cli.py

```diff
@@ -159,6 +168,29 @@ def run(
 
         obj.echo("Run link: {kfp_run_url}\n".format(kfp_run_url=kfp_run_url), fg="cyan")
 
+        if wait_for_completion:
+            response = flow._client.wait_for_run_completion(
+                run_pipeline_result.run_id, 500
+            ).to_dict()
+            if response["run"]["status"] == "Succeeded":
+                obj.echo(
+                    "Flow: {flow_name}, run link: {kfp_run_url}\n  SUCCEEDED!".format(
+                        flow_name=current.flow_name, kfp_run_url=kfp_run_url
+                    ),
+                    fg="green",
+                )
+                print("start_marker|success|end_marker")
+            else:
+                obj.echo(
+                    "Flow: {flow_name}, run link: {kfp_run_url}\n  FAILED!".format(
+                        flow_name=current.flow_name, kfp_run_url=kfp_run_url
+                    ),
+                    fg="red",
+                )
+                print("start_marker|failure|end_marker")
```

Comment:
Add exit code if wait for completion fails.

Reply:
Unfortunately, the KFP python API doesn't provide error messages of the run within KFP. It provides an error if the client (in this case Gitlab) cannot connect to the KF server and retrieve run details. See [here](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.server_api.html#kfp_server_api.models.ApiRun)
For now, I believe raising an exception here would suffice.